index.js

```diff
@@ -0,0 +1,30 @@
+const defaultMap = require('./default_map');
+
+const coreFields = [
+  'v',
+  'level',
+  'name',
+  'hostname',
+  'pid',
+  'time',
+  'msg',
+  'src'
+];
+
+module.exports = function bunyanEncoder(mapCoreFields = defaultMap, stream = process.stdout) {
+  return {
+    write: function (rec) {
+      if (typeof(rec) === 'string' || (rec instanceof String)) {
+        throw new Error('Stream must be raw');
+      }
+
+      const recCopy = JSON.parse(JSON.stringify(rec));
```

Comment:
Why are we copying the rec object? What stops us from modification of original record? I think it could be faster and more efficient in terms of using memory.